index.js

```diff
@@ -60,28 +60,24 @@ Level.prototype._del = function(id, options, callback) {
 }
 
 Level.prototype._put = function (key, value, options, callback) {
-  // TODO: do we still need to support ArrayBuffer?
-  if (value instanceof ArrayBuffer) {
-    value = Buffer.from(value)
-  }
-  var obj = this.convertEncoding(key, value, options)
-  this.idb.put(obj.key, obj.value, function() { callback() }, callback)
+  // TODO: once we upgrade abstract-leveldown, it will call _serializeValue for us.
+  value = this._serializeValue(value, options)
+  this.idb.put(key, value, function() { callback() }, callback)
 }
 
-Level.prototype.convertEncoding = function(key, value, options) {
-  if (options.raw) return {key: key, value: value}
-  if (value) {
-    var stringed = value.toString()
-    if (stringed === 'NaN') value = 'NaN'
-  }
-  var valEnc = options.valueEncoding
-  var obj = {key: key, value: value}
-  if (value && (!valEnc || valEnc !== 'binary')) {
-    if (typeof obj.value !== 'object') {
-      obj.value = stringed
-    }
-  }
-  return obj
+// NOTE: doesn't match abstract signature yet (which has no options argument).
+Level.prototype._serializeValue = function (value, options) {
+  // TODO: do we still need to support ArrayBuffer?
+  if (value instanceof ArrayBuffer) return Buffer.from(value)
+  if (value == null) return ''
+
+  // TODO: remove
+  if (options.raw) return value
+
+  // TODO: remove
+  if (typeof value !== 'object') return value.toString()
```

Comment:
[Similar to `memdown`](https://github.com/Level/memdown/blob/ee4602b570dc2461f476f3d19b9090d062f5d3a5/memdown.js#L140), in the next stage, `level-js` can just return the value as-is, without conversion.